editor/components/editor-history/undo.js

```diff
@@ -26,11 +18,11 @@ function EditorHistoryUndo( { hasUndo, undo } ) {
 	);
 }
 
-export default connect(
-	( state ) => ( {
-		hasUndo: hasEditorUndo( state ),
-	} ),
-	( dispatch ) => ( {
-		undo: () => dispatch( { type: 'UNDO' } ),
-	} )
-)( EditorHistoryUndo );
+export default compose( [
+	withSelect( ( select ) => ( {
+		hasUndo: select( 'core/editor' ).hasEditorUndo(),
+	} ) ),
+	withDispatch( ( dispatch ) => ( {
+		undo: () => dispatch( 'core/editor' ).undo(),
```

Comment:
Maybe we can apply the same as suggested for redo https://github.com/WordPress/gutenberg/pull/6167/files#r181509563.